docs/proposals/20220105-observatorium-alerting.md

```diff
@@ -0,0 +1,169 @@
+# Observatorium Alerting
+
+* **Owners:**
+  * `@onprem` `@bwplotka`
+
+* **Related Tickets:**
+  * https://issues.redhat.com/browse/MON-1761
+
+* **Other docs:*
+  * None 
+
+> TL;DR: For monitoring use cases, especially tenants of the metric signal of Observatorium, users want to be able to configure Prometheus-based alerts that will be evaluated for a given interval. If triggered they should notify the desired target (receiver) (e.g PagerDuty using Alertmanager Configuration). Users should also be able to see the status of all alerts through Prometheus-like alerts HTTP API. Users should also be able to see the status of all alerts through Prometheus-like Alerts HTTP API. This proposal specifies how such a system can be deployed within Observatorium.
+
+## Why
+
+Alerting is a critical component of reliable monitoring. We can build automation or notify humans in a reactive or proactive manner instead of watching dashboards.
+
+### Pitfalls of the current solution
+
+Observatorium does not have alerting support.
+
+## Goals
+
+Definition: 
+* Alert Routing: Configuration of Alertmanager related to receiver target config and its routing. 
+
+Goals:
+
+* Tenants should be able to read the status of active alerts 
+  * via ([Prometheus Alert API GET](https://prometheus.io/docs/prometheus/latest/querying/api/#alerts))
+* Tenants are able to configure alert routing:
+  * via new GET, DELETE and PUT HTTP API
+  * Potentially work with Alertmanager Team on upstream API.
+* Both alerts and alerting rules should be multi-tenant.
+
+## Non-Goals
+
+* Tenant should be able to create and edit alerting rules:
+  * Alerting Rules via GET and PUT via HTTP (Rules API)
+  * This is not the goal, because it should be done when recording rule API is done.
+* UI for Alerts/Rules, Alert Status, AM Routing, AM Alerts …
+* Alertmanager Alerts API, Silences, Inhibits API
+  * We should focus on that in next iteration.
+* Scaling Rulers
+* Different configuration than Alertmanager with 2 replicas.
+* Testing routing configuration or UI allowing better experience for configuring those. It is a common issue that routing can be very misleading. 
+
+## Abbreviations
+
+AM: Alertmanager
+
+## Audience of this proposal
+
+Observatorium Devs and Users.
+
+## How
+
+Within work for [Rules API proposal](https://observatorium.io/docs/proposals/20201019-prometheus-rules-in-observatorium-api.md/) we already added Rules APIs for reading and configuring rules which include both recording and alerting rules.
+
```

Comment:
Should we add a subsection here, something like `### Current solution`, it got me a bit confused since I thought this was already describing proposed solution.

Reply:
cc `git fetch origin pull453/head:<local branch name>`